pkg/model/packet_record.go

```diff
@@ -28,25 +28,25 @@ func NewPacketRecord(
 	return &pr
 }
 
-// ReadRawPacket reads a PacketRecord from a binary source, in LittleEndian order
+// ReadRawPacket reads a PacketRecord from a binary source, in NativeEndian order
 func ReadRawPacket(reader io.Reader) (*PacketRecord, error) {
 	var pr PacketRecord
 	currentTime := time.Now()
 	monotonicTimeNow := monotime.Now()
 	getLen := make([]byte, 4)
 	packetTimestamp := make([]byte, 8)
-	// Read IfIndex and discard it: To be used in other usecases
-	_ = binary.Read(reader, binary.LittleEndian, make([]byte, 4))
-	// Read Length of packet
-	_ = binary.Read(reader, binary.LittleEndian, getLen)
-	pr.Stream = make([]byte, binary.LittleEndian.Uint32(getLen))
-	// Read TimeStamp of packet
-	_ = binary.Read(reader, binary.LittleEndian, packetTimestamp)
+	// Read IfIndex and discard it: To be used in other use cases
+	_ = binary.Read(reader, binary.NativeEndian, make([]byte, 4))
+	// Read Length of a packet
+	_ = binary.Read(reader, binary.NativeEndian, getLen)
+	pr.Stream = make([]byte, binary.NativeEndian.Uint32(getLen))
+	// Read TimeStamp of a packet
+	_ = binary.Read(reader, binary.NativeEndian, packetTimestamp)
 	// The assumption is monotonic time should be as close to time recorded by ebpf.
 	// The difference is considered the delta time from current time.
-	tsDelta := time.Duration(uint64(monotonicTimeNow) - binary.LittleEndian.Uint64(packetTimestamp))
+	tsDelta := time.Duration(uint64(monotonicTimeNow) - binary.NativeEndian.Uint64(packetTimestamp))
 	pr.Time = currentTime.Add(-tsDelta)
 
-	err := binary.Read(reader, binary.LittleEndian, &pr.Stream)
+	err := binary.Read(reader, binary.NativeEndian, &pr.Stream)
```

Comment:
I get that the reads of other fields should be `NativeEndian`...

But if you are reading data from `skb->data`, shouldn't you be reading in `binary.BigEndian` format? `NetworkEndian` == `BigEndian`

Reply:
its array of bytes so it shouldn't matter but when copied to userspace we need to be in host endian fmt ?